### YamlMime:PythonClass
uid: azure.cognitiveservices.knowledge.qnamaker.runtime.models.ErrorResponseError
name: ErrorResponseError
fullName: azure.cognitiveservices.knowledge.qnamaker.runtime.models.ErrorResponseError
module: azure.cognitiveservices.knowledge.qnamaker.runtime.models
inheritances:
- azure.cognitiveservices.knowledge.qnamaker.runtime.models.error_py3.Error
summary: 'The error object.


  All required parameters must be populated in order to send to Azure.'
constructor:
  syntax: 'ErrorResponseError(*, code, message: str = None, target: str = None, details=None,
    inner_error=None, **kwargs) -> None'
  parameters:
  - name: code
    description: 'Required. One of a server-defined set of error codes.

      Possible values include: ''BadArgument'', ''Forbidden'', ''NotFound'',

      ''KbNotFound'', ''Unauthorized'', ''Unspecified'', ''EndpointKeysError'',

      ''QuotaExceeded'', ''QnaRuntimeError'', ''SKULimitExceeded'',

      ''OperationNotFound'', ''ServiceError'', ''ValidationFailure'',

      ''ExtractionFailure'''
    types:
    - <xref:str>
    - <xref:azure.cognitiveservices.knowledge.qnamaker.runtime.models.ErrorCodeType>
  - name: message
    description: A human-readable representation of the error.
    types:
    - <xref:str>
  - name: target
    description: The target of the error.
    types:
    - <xref:str>
  - name: details
    description: 'An array of details about specific errors that led to this

      reported error.'
    types:
    - <xref:list>[<xref:azure.cognitiveservices.knowledge.qnamaker.runtime.models.Error>]
  - name: inner_error
    description: 'An object containing more specific information than

      the current object about the error.'
    types:
    - <xref:azure.cognitiveservices.knowledge.qnamaker.runtime.models.InnerErrorModel>
